Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -42,17 +57,191 @@ public async Task<SearchGameSessionsResponse> SearchGameSessions(SearchGameSessi
+        }
+
+        private RetriveAwsCredentialsResponse SetupCredentials()
+        {
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+            var credentials = RetrieveAwsCredentials(_gameLiftClientSettings.ProfileName);
+            return credentials;
+        }
+
+        private async Task GenerateAuthToken()
         {
-            return new AmazonGameLiftClient(s_dummyAwsAccessKey, s_dummyAwsSecretKey, new AmazonGameLiftConfig
+            try
             {
-                ServiceURL = localEndpoint
-            });
+                var computeAuthTokenRequest = new GetComputeAuthTokenRequest
+                {
+                    ComputeName = _gameLiftClientSettings.ComputeName,
+                    FleetId = _gameLiftClientSettings.FleetID
+                };
+                var computeAuthTokenResponse = await GetComputeAuthToken(computeAuthTokenRequest);
+                _gameLiftClientSettings.AuthToken = computeAuthTokenResponse.AuthToken;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
         }
 
-        public async Task<DescribeGameSessionsResponse> DescribeGameSessions(DescribeGameSessionsRequest request)
+        private async Task RegisterCompute()
         {
-            return await _amazonGameLiftClient.DescribeGameSessionsAsync(request);
+            try
+            {
+                var ipAddress = Dns.GetHostEntry(Dns.GetHostName())
+                    .AddressList
+                    .FirstOrDefault(ip => ip.AddressFamily == AddressFamily.InterNetwork)?
+                    .ToString() ?? "0.0.0.0";
+                
+                var registerComputeRequest = new RegisterComputeRequest()
+                {
+                    ComputeName = _gameLiftClientSettings.ComputeName,
+                    FleetId = _gameLiftClientSettings.FleetID,
+                    IpAddress = ipAddress,
+                    Location = _gameLiftClientSettings.FleetLocation
+                };
+                var registerComputeResponse = await RegisterCompute(registerComputeRequest);
+                    
+                _gameLiftClientSettings.WebSocketUrl = registerComputeResponse.Compute.GameLiftServiceSdkEndpoint;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
         }
+
+        private async Task CreateCustomLocation()
+        {
+            try
+            {
+                ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+                var listLocationsResponse = await ListLocations(new ListLocationsRequest
+                {
+                    Filters = new List<string>{ "CUSTOM" }
+                });
+                
+                var foundLocation = listLocationsResponse.Locations.FirstOrDefault(l => l.LocationName.ToString() == _gameLiftClientSettings.FleetLocation);
```

Comment:
Please rename the method to indicate this behavior. Maybe `CreateCustomLocationIfNotExists`? or maybe change the call pattern to invoke `CreateCustomLocation` after checking if that location exists or not. I personally like to understand what a method is doing without having to go through the entire logic, and in this particular case its not evident to me that this method also does a duplicate check

Reply:
The ClientWrapper should only involve itself with accepting parameters (inputs) and invoking a corresponding SDK. The invoker should decide whether or not the location being used is coming from the _gameLiftClientSettings or something else. Can we isolate the wrapper from the _gameLiftClientSettings by following this pattern? This applies to all methods being added in